### DISCUSSION TITLE
`next-mdx-remote` Alternatives

### DISCUSSION THREAD

**Discussion by dstaley** [hashicorp/next-mdx-remote#438] [Show and tell]
We want to highlight alternatives to `next-mdx-remote` (particularly those with more features) in the README. So please share them here!

**Comment by talatkuyuk** [comment]
Hi @dstaley,

As I mentioned in the main [discussion](https://github.com/hashicorp/next-mdx-remote/discussions/437#discussioncomment-8648417), I wanted to share the alternative solution here.

The **[`next-mdx-remote-client`](https://github.com/ipikuka/next-mdx-remote-client)** serves as a **viable alternative** to **`next-mdx-remote`** having **more features**.

> [!NOTE]
> I appended the term "client" to the package name in order to emphasize for both server side and client side components which are exported by  **`next-mdx-remote-client`**, tailored for the "app" router and the "pages" router respectively.

| Features                                                    | `next-mdx-remote`   | `next-mdx-remote-client` |
| :---------------------------------------------------------- | :-----------------: | :----------------------: |
| support MDX version 3                                       | ✅               | ✅                    |
| ensure internal error handling mechanism in `app` router    | ❌                  | ✅                        |
| ensure internal error handling mechanism in `pages` router  | ❌                  | ✅                        |
| support export-from-MDX in `app` router                     | ❌                  | ✅                        |
| support export-from-MDX in `pages` router                   | ❌                  | ✅                        | 
| support import-into-MDX in `app` router                     | ❌                  | ✅                        |
| support import-into-MDX in `pages` router                   | ❌                  | ❌                        |
| get frontmatter and mutated scope in `app` router           | ❌                  | ✅                        |
| get frontmatter and mutated scope in `pages` router         | ✅                  | ✅                        |
| support options for disabling imports and exports in MDX    | ✅                  | ✅                        |
| support passing `vfile.data` into the `scope`               | ❌                  | ✅                        |
| provide utility for getting frontmatter without compiling   | ❌                  | ✅                        |
| expose `MDXProvider` from `@mdx-js/mdx`                     | ❌                  | ✅                        |
| provide option for disabling parent `MDXProvider` contexts  | ❌                  | ✅                        |
| expose the necessary types from `mdx/types`                 | ❌                  | ✅                        |

I published the **`next-mdx-remote-client`** on npm as [version 1.0.0](https://www.npmjs.com/package/next-mdx-remote-client). 

A demo app source code --> https://github.com/talatkuyuk/demo-next-mdx-remote-client

**Reply by arunavo4** [reply]
@talatkuyuk Thanks for doing this, this has one of the most complete and easy to follow docs. Even a migration guide that saves a lot of time.

**Reply by talatkuyuk** [reply]
Thank you @arunavo4; your positive comment is so valuable ❤️ 🙏, which is going to lead the developers to have a look and try **`next-mdx-remote-client`**.

**Comment by devjiwonchoi** [comment]
## link: [remote-mdx](https://github.com/devjiwonchoi/remote-mdx)

Hello engineers, just landed `remote-mdx` with the following benefits:

- Package Size: **Reduced 90%** of the original package size.
- Fully Typed: Does not depend on `allowJS` or `any` type.
- Module: Native ESM, no internal `commonjs` resolution that breaks `next dev --turbo`.

Replace the current `next-mdx-remote` by:

> Example of using the package manager `pnpm`.

```sh
pnpm add remote-mdx
pnpm remove next-mdx-remote
```

It is a drop-in replacement of `next-mdx-remote`:

```diff
- import { MDXRemote } from 'next-mdx-remote/rsc'
+ import { MDXRemote } from 'remote-mdx/rsc'
```

I hope this package can be of assistance in improving `next-mdx-remote`.
Feel free to [open an issue here](https://github.com/devjiwonchoi/remote-mdx/issues) so that I can fix it ASAP for you.
